packages/protocol/test/unit/utils/StreamPartitionID.test.ts

```diff
@@ -0,0 +1,188 @@
+import SPID  from '../../../src/utils/StreamPartitionID'
+
+describe('StreamPartitionID', () => {
+    const STREAM_ID = 'test-stream-id'
+    const PARTITION = 1
+
+    it('can use object destructuring', () => {
+        const spid = new SPID(STREAM_ID, PARTITION)
+        const { id, partition, key } = spid
+        expect(id).toBe(STREAM_ID)
+        expect(partition).toBe(PARTITION)
+        expect(key).toBe(spid.toString())
+    })
+
+    it('defaults partition to 0', () => {
+        const spid = new SPID(STREAM_ID)
+        expect(spid.id).toEqual(STREAM_ID)
+        expect(spid.partition).toBe(0)
+    })
+
+    it('lowercases streamId', () => {
```

Comment:
As I expressed in an earlier comment, I believe the intention currently is not to touch the casing at all.

Reply:
done